app/Http/Controllers/Settings/UserRoleAssignController.php

```
@@ -0,0 +1,62 @@
+        ]);
+    }
+
+    public function update(Request $request, User $user): RedirectResponse
+    {
+        $validated = $request->validate([
+            'user' => ['required', 'exists:users,id'],
+            'role' => ['required', 'string', 'exists:roles,name'],
+            'permissions' => ['required', 'array'],
+        ]);
+
+        $user->syncRoles([$validated['role']]);
```

PHPStan (2.0.3)

Cannot access offset 'role' on mixed.


(offsetAccess.nonOffsetAccessible)